src/raftstore/store/snap.rs

```diff
@@ -1985,9 +1986,17 @@ impl SnapManager {
                     error!("failed to parse snapkey from {}", name);
                     return None;
                 }
-                Some((SnapKey::new(numbers[0], numbers[1], numbers[2]), is_sending))
+                let snap_key = SnapKey::new(numbers[0], numbers[1], numbers[2]);
+                if core.registry.contains_key(&snap_key) {
```

Comment:
I don't think if a snapshot is not registered, then it's idle. For example, a snapshot is generated and pending for sending out. It's not registered, but it's in used. Another example, a snapshot is received from remote and stored to disk. But it's not applied yet. It's not registered but it's in used. If delete it in this case, it probably leads to crash and can be recovered automatically.